src/Microsoft.ComponentDetection.Detectors/npm/NpmComponentDetector.cs

```diff
@@ -81,16 +84,80 @@ protected virtual bool ProcessIndividualPackageJTokens(string filePath, ISingleF
         {
             var name = packageJToken["name"].ToString();
             var version = packageJToken["version"].ToString();
-
+            var authorToken = packageJToken["author"];
+            
             if (!SemanticVersion.TryParse(version, out _))
             {
                 Logger.LogWarning($"Unable to parse version \"{version}\" for package \"{name}\" found at path \"{filePath}\". This may indicate an invalid npm package component and it will not be registered.");
                 return false;
             }
 
-            var detectedComponent = new DetectedComponent(new NpmComponent(name, version));
+            var npmComponent = new NpmComponent(name, version);
+            npmComponent.Author = GetAuthor(authorToken, name, filePath);
+
+            var detectedComponent = new DetectedComponent(npmComponent);
+            
             singleFileComponentRecorder.RegisterUsage(detectedComponent);
             return true;
         }
+
+        private NpmAuthor GetAuthor(JToken authorToken, string packageName, string filePath)
+        {
+            var authorString = authorToken?.ToString();
+            if (string.IsNullOrEmpty(authorString))
+            {
+                return null;
+            }
+
+            string authorName = null;
+            string authorEmail = null;
+
+            //for parsing author in Json Format
+            // for e.g. 
+            // "author": {
+            //     "name": "John Doe",
+            //     "email": "johndoe@outlook.com",
+            //     "name": "https://jd.com",
+            // }
+            if (authorString.StartsWith("{"))
+            {
+                authorName = authorToken["name"]?.ToString();
+                authorEmail = authorToken["email"]?.ToString();
+            }
+
+            // for parsing single line author.
+            // for e.g. "author": "John Doe <johnDoe@outlook.com> (https://jd.com)"
+            else
+            {
+                bool isEmailPresent = authorString.Contains("<");
```

Comment:
I would be wary of such a simple validation here. Expecting the data to be well formed may not always be accurate and trying to exact match may lead to unexpected errors getting thrown.

Reply:
introduced JSON  and regex-based verification of author.